### ISSUE TITLE
Tolerate <main role="main">

### ISSUE THREAD

**Issue by stonio** [validator/validator#386]
This HTML code generates a WARNING _Element `main` does not need a `role` attribute_.

``` html
<!DOCTYPE html>
<html>
<head>
<title>Test</title>
</head>
<body>
<main role="main"></main>
</body>
</html>
```

However, [MDN](https://developer.mozilla.org/en/docs/Web/HTML/Element/main#Browser_compatibility) recommends to use this role for old browser compatibility.

_Nu Html Checker_ should tolerate `role="main"` attribute on element `main`.

**Comment by stevefaulkner** [comment]
@stonio The validator implements its ARIA usage rules based on the following spec https://www.w3.org/TR/html-aria/#docconformance

Note it is not prohibited (not an error) to add default implicit roles such as `<main role=main>`, but it is discouraged (thus the warning).

MDN states a suggestion that was, but is no longer in the [HTML spec](http://w3c.github.io/html/grouping-content.html#the-main-element).

the following may be helpful in understanding why the addition of default implicit semantics is discouraged: 
[On HTML belts and ARIA braces (The Default Implicit ARIA semantics they didn’t want you to know about) ](http://html5doctor.com/on-html-belts-and-aria-braces/)

**Comment by AndySky21** [comment]
For what I see, this is flagged as a warning and not as an error.

In HTML [5.1 PR](https://www.w3.org/TR/html/grouping-content.html#the-main-element) and [5.2 WD](http://w3c.github.io/html/grouping-content.html#the-main-element) `main` role is listed as "default - do not set" as opposed to `presentation`. 
According to ARIA authoring requirements (in both versions):

> In the majority of cases setting an ARIA `role` and/or `aria-*` attribute that matches the default implicit ARIA semantics is unnecessary and not recommended as these properties are already set by the browser.

So I guess validator is right in warning authors against it. It's up to the authors to decide whether or not legacy compatibility is a strong requirement for their own projects.

**Comment by stevefaulkner** [comment]
@AndySky21 note the info in the HTML spec is informative, HTML references ARIA in HTML normatively.

**Comment by AndySky21** [comment]
The funny part is that the article on HTMLdoctor references the note in HTML spec too :D
I often end up quoting even non-normative parts of HTML spec because they are based on normative references and it's more likely authors are more familiar with them, rather than sources - however I know it's a limited approach.

**Comment by sideshowbarker** [comment]
Closing since as https://github.com/validator/validator/issues/386#issuecomment-254206980 explains the checker is behaving as expected per-spec here.
